app/src/common/shared/com/igalia/wolvic/ui/viewmodel/WindowViewModel.java

```diff
@@ -259,6 +266,7 @@ public void onChanged(ObservableBoolean o) {
                         UrlUtils.isFileUri(aUrl) ||
                         UrlUtils.isHomeUri(getApplication(), aUrl) ||
                         isLibraryVisible.getValue().get() ||
+                        isDownloadsVisible.getValue().get() ||
```

Comment:
Actually, if we use `isWebContentVisible` for all, it can be extremely easy for us to have race conditions for setting the values of `isWebContentVisible` when e.g. switching from downloads to library, which can cause many issues and that is why I decided to do this way and I think we should continue doing this.

Anyway we have a similar method as you proposed to call at https://github.com/Igalia/wolvic/pull/881#discussion_r1405942102 And I can see you have already noticed that.